### PR TITLE
Add support for images optimized for dark themes

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Images with dark text and transparent backgrounds can be hard to see when the image is embedded in a dark theme, those images often have a variant that is easier to look at in that setting.
This PR allows requesting a `dark_` prefixed image of all types if it exists.

If this PR is accepted, I will be submitting a PR to the frontend so it can be used.

logos for `uptimerobot` optimized for dark themes have been added.

## Type of change

- [ ] Add a new logo or icon for a new core integration
- [ ] Add a missing icon or logo for an existing core integration
- [ ] Add a new logo or icon for a custom integration (custom component)
  - [ ] I've opened up a PR for my custom integration on the [Home Assistant
    Python wheels repository](https://github.com/home-assistant/wheels-custom-integrations)
- [ ] Replace an existing icon or logo with a higher quality version
- [ ] Removing an icon or logo

## Additional information

- This PR fixes or closes issue: fixes #
- Link to code base pull request: 
- Link to documentation pull request: 
- Link to integration documentation on our website: 

## Checklist

- [x] The added/replaced image(s) are **PNG**
- [ ] Icon image size is 256x256px (`icon.png`)
- [ ] hDPI icon image size is 512x512px for  (`icon@2x.png`)
- [x] Logo image size has min 128px, but max 256px, on the shortest side (`logo.png`)
- [x] hDPI logo image size has min 256px, but max 512px, on the shortest side (`logo@2x.png`)

### CODE REVIEW

**Comment by frenck** [comment]
I'm ok with this, as in, would solve some issues 👍 

Should we extend the validation to ensure the dark variants are conformed to specifications as well?

<https://github.com/home-assistant/brands/blob/master/.github/workflows/validate/validate.sh>

**Comment by ludeeus** [comment]
> Should we extend the validation to ensure the dark variants are conformed to specifications as well?

That's needed and added :+1:

**Review by frenck — APPROVED** [review]
LGTM ✅

**Comment by frenck** [comment]
1 Small task left: Could we update the README of the repo? :)

**Comment by frenck** [comment]
LOL 🤦  I've my head dead today.

File: scripts/build.sh [hunk 1]
```diff
@@ -38,6 +38,16 @@ find ./build -type f -name "logo.png" | while read logo; do
   fi
 done
 
+# Create fallback for dark variants
+find ./build -type f -type f -name "icon.png" -o -name "icon@2x.png" -o -name "logo.png" -o -name "logo@2x.png" | while read image; do
+  dir=$(dirname "${image}")
+  filename=$(basename -s .png "${image}")
+  if [[ ! -f "${dir}/dark_${filename}.png" ]]; then
+    cp "${image}" "${dir}/dark_${filename}.png"
```
Comment (bramkragten) [inline scripts/build.sh:46]: Not sure if we should or if it would be better, but couldn't we use rewrites instead of copying the images?

Comment (ludeeus) [reply scripts/build.sh:46]: I guess we could?

**Pull request merged**